internal/core/metadata/rest_deviceprofile.go

```diff
@@ -102,6 +112,8 @@ func restUpdateDeviceProfile(w http.ResponseWriter, r *http.Request) {
 				http.Error(w, err.Error(), http.StatusNotFound)
 			case *errors2.ErrValueDescriptorsInUse:
 				http.Error(w, err.Error(), http.StatusConflict)
+			case errors.ErrDeviceProfileInvalidState:
+				http.Error(w, err.Error(), http.StatusConflict)
```

Comment:
I also notice here the difference in referring to error types with pointers versus not. In truth, these don't have to be pointer references so I'm aligned to the logic that's been added. But we should have an issue to address this globally and standardize.